open_feature/open_feature_client.py

```diff
@@ -152,27 +166,65 @@ def evaluate_flag_details(
         evaluation_context: EvaluationContext = None,
         flag_evaluation_options: typing.Any = None,
     ) -> FlagEvaluationDetails:
+        """
+        Evaluate the flag requested by the user from the clients provider.
+
+        :param flag_type: the type of the flag being returned
+        :param key: the string key of the selected flag
+        :param default_value: backup value returned if no result found by the provider
+        :param evaluation_context: Information for the purposes of flag evaluation
+        :param flag_evaluation_options: Additional flag evaluation information
+        :return: a FlagEvaluationDetails object with the fully evaluated flag from a
+        provider
+        """
+
         if evaluation_context is None:
             evaluation_context = EvaluationContext()
+
+        hook_context = HookContext(
+            flag_key=key,
+            flag_type=flag_type,
+            default_value=default_value,
+            evaluation_context=evaluation_context,
+            client_metadata=None,
+            provider_metadata=None,
+        )
+        merged_hooks = []
+
         try:
-            # The merge below will eventually pull in Evaluation contexts from Hooks
-            invocation_context = EvaluationContext()
+            invocation_context = before_hooks(
```

Comment:
I'm a bit confused by the use of the word `invocation_context` here but then `merged_context` below. Is it worth adding some comments to explain a little about what's happening with the different context objects and why?

Reply:
Comments have been added to the code and a link to the spec